libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -514,70 +517,105 @@ protected async Task<DialogTurnResult> ContinueActionsAsync(DialogContext dc, ob
+                    var result = await actionDC.ContinueDialogAsync(cancellationToken).ConfigureAwait(false);
 
-                // Is the step waiting for input or were we cancelled?
-                if (result.Status == DialogTurnStatus.Waiting || GetUniqueInstanceId(actionContext) != instanceId)
-                {
-                    return result;
-                }
+                    // Start step if not continued
+                    if (result.Status == DialogTurnStatus.Empty && GetUniqueInstanceId(actionContext) == instanceId)
+                    {
+                        // Call begin dialog on our next step, passing the effective options we computed
+                        var nextAction = actionContext.Actions.First();
+                        result = await actionDC.BeginDialogAsync(nextAction.DialogId, nextAction.Options, cancellationToken).ConfigureAwait(false);
+                    }
 
-                // End current step
-                await EndCurrentActionAsync(actionContext, cancellationToken).ConfigureAwait(false);
+                    // Is the step waiting for input or were we cancelled?
+                    if (result.Status == DialogTurnStatus.Waiting || GetUniqueInstanceId(actionContext) != instanceId)
+                    {
+                        return result;
+                    }
 
-                if (result.Status == DialogTurnStatus.CompleteAndWait)
-                {
-                    // Child dialog completed, but wants us to wait for a new activity
-                    result.Status = DialogTurnStatus.Waiting;
-                    return result;
-                }
+                    // End current step
+                    await EndCurrentActionAsync(actionContext, cancellationToken).ConfigureAwait(false);
 
-                var parentChanges = false;
-                DialogContext root = actionContext;
-                var parent = actionContext.Parent;
-                while (parent != null)
-                {
-                    var ac = parent as ActionContext;
-                    if (ac != null && ac.Changes != null && ac.Changes.Count > 0)
+                    if (result.Status == DialogTurnStatus.CompleteAndWait)
                     {
-                        parentChanges = true;
+                        // Child dialog completed, but wants us to wait for a new activity
+                        result.Status = DialogTurnStatus.Waiting;
+                        return result;
                     }
 
-                    root = parent;
-                    parent = root.Parent;
-                }
+                    var parentChanges = false;
+                    DialogContext root = actionContext;
+                    var parent = actionContext.Parent;
+                    while (parent != null)
+                    {
+                        var ac = parent as ActionContext;
+                        if (ac != null && ac.Changes != null && ac.Changes.Count > 0)
```

Comment:
I prefer `ac.Changes.Any()` instead of `ac.Changes.Count > 0` but not a big deal, up to you